test/common/router/router_ratelimit_test.cc

```diff
@@ -552,6 +552,61 @@ TEST_F(RateLimitPolicyEntryTest, HeaderValueMatchNoMatch) {
   EXPECT_TRUE(descriptors_.empty());
 }
 
+TEST_F(RateLimitPolicyEntryTest, HeaderValueMatchHeadersNotPresent) {
+  std::string json = R"EOF(
+  {
+    "actions": [
+      {
+        "type": "header_value_match",
+        "descriptor_value": "fake_value",
+        "expect_match": false,
+        "headers": [
+          {
+            "name": "x-header-name",
+            "value": "test_value",
+            "regex": false
+          }
+        ]
+      }
+    ]
+  }
+  )EOF";
+
+  SetUpTest(json);
+  Http::TestHeaderMapImpl header{{"x-header-name", "fake_value"}};
```

Comment:
Changing the header value to `not_same_value` or something similar would make this test more clear, i.e the value is not the same as the descriptor_value, and what is loaded to the descriptors is the descriptor value.